File: client/src/app/modules/tiers/services/jss.subscription.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { JssSubscriptionType } from '../../tiers/model/JssSubscriptionType';

@Injectable({
  providedIn: 'root'
})
export class JssSubscriptionTypeService extends AppRestService<JssSubscriptionType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getJssSubscriptionTypes() {
    return this.getList(new HttpParams(), "jss-suscription-types");
  }

}

File: src/main/java/com/jss/jssbackend/modules/tiers/service/JssSubscriptionTypeService.java
package com.jss.jssbackend.modules.tiers.service;

import java.util.List;

import com.jss.jssbackend.modules.tiers.model.JssSubscriptionType;

public interface JssSubscriptionTypeService {
    public List<JssSubscriptionType> getJssSubscriptionTypes();

    public JssSubscriptionType getJssSubscriptionType(Integer id);
}
